### PR TITLE
optimize ZCS file containment check by using file sections

### PR DESCRIPTION
**Pull request by jasonstack**

### CODE REVIEW

File: src/java/org/apache/cassandra/db/streaming/CassandraOutgoingFile.java [hunk 1]
```diff
@@ -185,46 +177,18 @@ public boolean shouldStreamEntireSSTable()
...
-            while (iter.hasNext())
-            {
-                DecoratedKey key = iter.next();
-                if (!rangeOwnHelper.check(key))
-                    return false;
-            }
-        }
-        return true;
+        // if transfer sections contain entire sstable
+        long transferLength = sections.stream().mapToLong(p -> p.upperPosition - p.lowerPosition).sum();
```
Comment (tjake) [inline src/java/org/apache/cassandra/db/streaming/CassandraOutgoingFile.java:190]: These sections should be normalized, so you will only have 1 when its the full sstable right?

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,10 +3,10 @@ jobs:
   j8_jvm_upgrade_dtests:
     docker:
     - image: spod/cassandra-testing-ubuntu1810-java11-w-dependencies:20190306
-    resource_class: medium
```
Comment (tjake) [inline .circleci/config.yml:6]: I assume you didn't intend to commit this...

**Review by tjake — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (jasonstack) [reply .circleci/config.yml:6]: right, not to commit. it's to speed up CI during our circle-ci trial period..

File: src/java/org/apache/cassandra/db/streaming/CassandraOutgoingFile.java (hunk 1 above)
Comment (jasonstack) [reply src/java/org/apache/cassandra/db/streaming/CassandraOutgoingFile.java:190]: didn't realize `Range.normalize` will combine overlapping ranges.. updated here 188ed6d

**Review by tjake — APPROVED** [review]

File: src/java/org/apache/cassandra/db/streaming/CassandraOutgoingFile.java (hunk 1 above)
Comment (jasonstack) [reply src/java/org/apache/cassandra/db/streaming/CassandraOutgoingFile.java:190]: I think there can be more than 1 section if ranges don't overlap..
eg. transfer ranges (0, 1000], (2000, 3000] and sstable has token 100, 200, 2100, 2200.. there will be two sections..

I have reverted to `long transferLength = sections.stream().mapToLong(p -> p.upperPosition - p.lowerPosition).sum();`  WDYT?

**Pull request closed without merging**